streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -502,6 +517,13 @@ public StreamsConfig(final Map<?, ?> props) {
         consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, originals().get(BOOTSTRAP_SERVERS_CONFIG));
         // remove deprecate ZK config
         consumerProps.remove(ZOOKEEPER_CONNECT_CONFIG);
+        if (eosEnabled) {
+            if (clientProvidedProps.containsKey(ConsumerConfig.ISOLATION_LEVEL_CONFIG)) {
```

Comment:
Because this is current default behavior. I agree we can log and we do have a JIRA and PR for this already (https://issues.apache.org/jira/browse/KAFKA-5096). Did not update this code yet accordingly...